Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -539,41 +445,46 @@ void CGPathAddEllipseInRect(CGMutablePathRef path, const CGAffineTransform* m, C
  @Status Interoperable
 */
 void CGPathCloseSubpath(CGMutablePathRef path) {
-    if (path == NULL) {
-        return;
+    // Move the current point to the starting point since the line is closed.
+    if (!CGPointEqualToPoint(path->_impl.currentPoint, path->_impl.startingPoint)) {
+        path->_impl.currentPoint = path->_impl.startingPoint;
     }
-
-    _CGPathAddElement(path, kCGPathElementCloseSubpath);
+    path->endFigure(D2D1_FIGURE_END_CLOSED);
 }
```

Comment:
Yes, it's fine to let D2D close the path since we moved it to a function that handles the bad case more easily of the path already being closed.

---
In reply to: [83963413](https://github.com/Microsoft/WinObjC/pull/1138#discussion_r83963413) [](ancestors = 83963413,83963372)